service/src/integrationTest/kotlin/fi/espoo/evaka/application/ApplicationControllerCitizenIntegrationTest.kt

```diff
@@ -0,0 +1,141 @@
+// SPDX-FileCopyrightText: 2017-2023 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.application
+
+import fi.espoo.evaka.FullApplicationTest
+import fi.espoo.evaka.insertApplication as insertTestApplication
+import fi.espoo.evaka.placement.PlacementType
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.auth.CitizenAuthLevel
+import fi.espoo.evaka.shared.auth.UserRole
+import fi.espoo.evaka.shared.dev.insertTestCareArea
+import fi.espoo.evaka.shared.dev.insertTestDaycare
+import fi.espoo.evaka.shared.dev.insertTestEmployee
+import fi.espoo.evaka.shared.dev.insertTestPerson
+import fi.espoo.evaka.shared.domain.BadRequest
+import fi.espoo.evaka.shared.domain.HelsinkiDateTime
+import fi.espoo.evaka.shared.domain.MockEvakaClock
+import fi.espoo.evaka.test.getApplicationStatus
+import fi.espoo.evaka.testAdult_1
+import fi.espoo.evaka.testArea
+import fi.espoo.evaka.testChild_1
+import fi.espoo.evaka.testDaycare
+import fi.espoo.evaka.testDecisionMaker_1
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.assertThrows
+import org.springframework.beans.factory.annotation.Autowired
+
+class ApplicationControllerCitizenIntegrationTest : FullApplicationTest(resetDbBeforeEach = true) {
+    @Autowired private lateinit var applicationControllerCitizen: ApplicationControllerCitizen
+
+    @Autowired private lateinit var stateService: ApplicationStateService
+
+    private val clock = MockEvakaClock(HelsinkiDateTime.now())
+
+    @BeforeEach
+    fun beforeEach() {
+        db.transaction { tx ->
+            tx.insertTestPerson(testAdult_1)
```

Comment:
This is much better than using `insertGeneralTestFixtures` but this test still depends on various global fixture definitions. Does it *really* depend on the actual values, or could we just use minimal `DevFoo(...)` data here?

Reply:
I think issue was inserting the application form which assumed to have those specific people. Long term fix could be to add a proper `insertTestApplication` function and deprecate/remove others. Currently we have many functions related to that, but none are really good. The one I ended up using even had a nameclash with the normal insert so I had to rename it in the import. I'll create a ticket to inbox.